core/sdk-core/src/test/java/software/amazon/awssdk/core/BatchOverrideConfigurationTest.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core;
+
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class BatchOverrideConfigurationTest {
```

Comment:
Let's add hashcode and equals to `BatchOverrideConfiguration`. `BatchManager` is not needed for now because it's internal/protected API and also not a pojo class

Reply:
Added hashcode and equals method to `BatchOverrideConfiguration`. Also added testing as well.